src/client/stratum.rs

```diff
@@ -169,11 +170,17 @@ impl StratumHandler {
     pub async fn connect(
         address: String,
         miner_address: String,
+        ssl_true: bool,
         mine_when_not_synced: bool,
         block_template_ctr: Option<Arc<AtomicU16>>,
     ) -> Result<Box<Self>, Error> {
         info!("Connecting to {}", address);
-        let socket = TcpStream::connect(address).await?;
+        
+        if ssl_true == true {
+            let socket = SslStream::connect(address).await?;
+        } else {
+            let socket = TcpStream::connect(address).await?;
+        }
```

Comment:
@splatboydev This part doesn't compile, right?

Reply:
Yes, that's correct.